crates/extensions/tedge_gen_mapper/src/actor.rs

```diff
@@ -34,11 +41,26 @@ impl Actor for GenMapper {
                 _ = interval.tick() => {
                     self.tick().await?;
                 }
-                message = self.mqtt.recv() => {
-                    match message.map(Message::try_from) {
-                        Some(Ok(message)) => self.filter(message).await?,
-                        Some(Err(err)) => {
-                            error!(target: "gen-mapper", "Cannot process message: {err}");
+                message = self.messages.recv() => {
+                    match message {
+                        Some(InputMessage::MqttMessage(message)) => match Message::try_from(message) {
+                            Ok(message) => self.filter(message).await?,
+                            Err(err) => {
+                                error!(target: "gen-mapper", "Cannot process message: {err}");
+                            }
+                        },
+                        Some(InputMessage::FsWatchEvent(FsWatchEvent::Modified(path))) => {
+                            let Ok(path) = Utf8PathBuf::try_from(path) else {
+                                continue;
+                            };
+                            if matches!(path.extension(), Some("js" | "ts")) {
+                                self.reload_filter(path).await;
+                            } else if path.extension() == Some("toml") {
+                                self.reload_toml(path).await;
+                            }
+                        },
+                        Some(InputMessage::FsWatchEvent(e)) => {
+                            tracing::warn!("TODO do something with {e:?}")
```

Comment:
We can definitely ignore anything that is is related to a file with an unknown extension (such as `.collectd.toml.swp` and `collectd.toml~`).

- Deletion of a pipeline should be okay, unless if we try to unsubscribe to the input topics.
- Deletion of a script is more problematic, if still used by one of the pipeline.